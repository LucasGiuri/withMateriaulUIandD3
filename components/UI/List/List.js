import PropTypes from 'prop-types';
import NavLink from '../NavLink/NavLink';
import { Grid } from './List.styles';
import Title from '../Title/Title';
import Item from '../Item/Item';

const List = ({list, filterBy, section}) => {
  const sectionName = section.toLowerCase();
  const filterName = filterBy.toLowerCase();

  return (
    <Grid>
      {list ? 
        list.map(item => {
          const index = item.url.indexOf(filterName) + filterName.length;
          const id = item.url.substring(index, item.url.length);

          return (
            <NavLink key={item.url} href={sectionName} as={`${sectionName}${id}`}>
              <Item>{item.name ? item.name : item.title}</Item>
            </NavLink>
          )
        }) : 
        <Title text={"No results Found :("} />
      }
    </Grid>
  )
};

List.propTypes = {
  list: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.string,
    name: PropTypes.string
  })),
  filterBy: PropTypes.string.isRequired,
  section: PropTypes.string.isRequired
};

export default List;