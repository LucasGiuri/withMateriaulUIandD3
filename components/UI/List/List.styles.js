import styled from 'styled-components';
import { BLACK } from '../../../public/styles/colors';

export const Grid = styled.div`
  background-color: ${BLACK};
  div {
    border-top: none;
  }
`;