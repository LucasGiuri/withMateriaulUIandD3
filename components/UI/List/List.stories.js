import React from 'react';
import { storiesOf } from '@storybook/react';
import List from './List';

const listOfNames = [{id:1, name: 'lucas'}, {id:2, name: 'messi'}];

storiesOf('List')
  .add('List default with empty list', () => (
    <List list={[]}/>
  ))
  .add('List default with complete list', () => (
    <List list={listOfNames}/>
  ))
