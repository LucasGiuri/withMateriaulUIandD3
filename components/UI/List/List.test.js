import React from 'react';
import renderer from 'react-test-renderer';
import List from './List';
import 'jest-styled-components';

const section = "FILMS";
const filterBy = "film";
const listWithName = [{
  url: "1",
  name: "item 1"
},
{
  url: "2",
  name: "item 2"
},
{
  url: "3",
  name: "item 3"
}];
const listWithTitle = [{
  url: "1",
  title: "item 1"
},
{
  url: "2",
  title: "item 2"
},
{
  url: "3",
  title: "item 3"
}];

describe('List', () => {
  it('List basic render snapshoot with empty list', () => {
    const component = renderer.create(<List list={[]} section={section} filterBy={filterBy} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })

  it('List complete with obj.name render snapshoot with empty list', () => {
    const component = renderer.create(<List list={listWithName} section={section} filterBy={filterBy} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })

  it('List complete with obj.title render snapshoot with empty list', () => {
    const component = renderer.create(<List list={listWithTitle} section={section} filterBy={filterBy} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })
})
