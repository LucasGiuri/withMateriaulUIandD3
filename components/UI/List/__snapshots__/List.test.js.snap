// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List List basic render snapshoot with empty list 1`] = `
.c0 {
  background-color: #000000;
}

.c0 div {
  border-top: none;
}

<div
  className="c0"
/>
`;

exports[`List List complete with obj.name render snapshoot with empty list 1`] = `
.c1 {
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: 'Roboto',sans-serif;
}

.c0 {
  background-color: #000000;
}

.c0 div {
  border-top: none;
}

.c2 {
  padding: 14px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  cursor: pointer;
  border-bottom: 1px solid #999999;
  font-family: 'Roboto',sans-serif;
  font-size: 22px;
  -webkit-transition: 0.25s;
  transition: 0.25s;
  color: #999999;
}

.c2:hover {
  color: #FFFFFF;
}

<div
  className="c0"
>
  <a
    className="c1"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    <div
      className="c2"
    >
      item 1
    </div>
  </a>
  <a
    className="c1"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    <div
      className="c2"
    >
      item 2
    </div>
  </a>
  <a
    className="c1"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    <div
      className="c2"
    >
      item 3
    </div>
  </a>
</div>
`;

exports[`List List complete with obj.title render snapshoot with empty list 1`] = `
.c1 {
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: 'Roboto',sans-serif;
}

.c0 {
  background-color: #000000;
}

.c0 div {
  border-top: none;
}

.c2 {
  padding: 14px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  cursor: pointer;
  border-bottom: 1px solid #999999;
  font-family: 'Roboto',sans-serif;
  font-size: 22px;
  -webkit-transition: 0.25s;
  transition: 0.25s;
  color: #999999;
}

.c2:hover {
  color: #FFFFFF;
}

<div
  className="c0"
>
  <a
    className="c1"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    <div
      className="c2"
    >
      item 1
    </div>
  </a>
  <a
    className="c1"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    <div
      className="c2"
    >
      item 2
    </div>
  </a>
  <a
    className="c1"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    <div
      className="c2"
    >
      item 3
    </div>
  </a>
</div>
`;
